### PR TITLE
Start of Sponge support

### PR DESCRIPTION
**Pull request by kashike**

### CODE REVIEW

File: sponge/src/main/java/co/aikar/commands/SpongeRootCommand.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    }
+
+    @Override
+    public boolean testPermission(CommandSource source) {
+        return true;
+    }
+
+    @Override
+    public Optional<Text> getShortDescription(CommandSource source) {
+        return Optional.empty();
```
Comment (kashike) [inline sponge/src/main/java/co/aikar/commands/SpongeRootCommand.java]: Not sure how to support this.

File: sponge/src/main/java/co/aikar/commands/SpongeRootCommand.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    }
+
+    @Override
+    public Optional<Text> getHelp(CommandSource source) {
+        return Optional.empty();
```
Comment (kashike) [inline sponge/src/main/java/co/aikar/commands/SpongeRootCommand.java]: Not sure how to support this.

File: sponge/src/main/java/co/aikar/commands/SpongeRootCommand.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    }
+
+    @Override
+    public Text getUsage(CommandSource source) {
+        return Text.of();
```
Comment (kashike) [inline sponge/src/main/java/co/aikar/commands/SpongeRootCommand.java]: Not sure how to support this.

File: sponge/src/main/java/co/aikar/commands/SpongeRootCommand.java (hunk 1 above)
Comment (aikar) [inline sponge/src/main/java/co/aikar/commands/SpongeRootCommand.java]: for root commands, pretty much useless for now. Gotta figure out #3

File: sponge/src/main/java/co/aikar/commands/SpongeRootCommand.java (hunk 2 above)
Comment (aikar) [inline sponge/src/main/java/co/aikar/commands/SpongeRootCommand.java]: same as above, #3

**Pull request merged**